More named inference rules for pointwise unary ops

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #22341 Add smoke test for information fn/method/attrs to test_namedtensor
* **#22308 More named inference rules for pointwise unary ops**

Featuring:
- digamma, erfinv, lgamma, polygamma, sigmoid, tanh, zero_

Test Plan:
- `python test/test_namedtensor.py -v` [namedtensor ci]

gh-metadata: pytorch pytorch 22308 gh/zou3519/65/head

Differential Revision: [D16053441](https://our.internmc.facebook.com/intern/diff/D16053441)

zou3519 committed:
More named inference rules for pointwise unary ops

Featuring:
- digamma, erfinv, lgamma, polygamma, sigmoid, tanh, zero_

Test Plan:
- `python test/test_namedtensor.py -v` [namedtensor ci]

nairbv commented on aten/src/THC/generic/THCTensorMathPointwise.cu:
```diff
@@ -209,7 +208,7 @@ static void propagate_names(THCTensor* result, THCTensor* src) {
                                                                         \
     THCudaCheck(cudaGetLastError());                                    \
-    propagate_names(self_, src);                                        \
+    propagate_names_if_named_tensor_enabled(self_, src);                \
```
if we're going to be adding more uses of this, we might want to trim off the `if_named_tensor_available` to reduce cleanup when it's fully on.

It seems clear to me without the `if`, but if you want an expressive name it could also just be something like `propagate_any_names` ... I think an unnamed tensor and !named_tensor_enabled will essentially be equivalent, and then at least you capture that it could end up being a no-op.

zou3519 replied:
There will still be cleanup when it's fully on, even if the name is only `propagate_names`. When we remove the `NAMEDTENSOR_ENABLED` macro, the `propagate_names` function will look like
```
void propagate_names(...) {
    at::namedinference::propagate_names(...);
}
```
which is silly so we get rid of it and replace it straight up with `at::namedinference::propagate_names`.

I think it's clearer right now as `propagate_names_if_named_tensor_enabled` because it refers to the NAMEDTENSOR_ENABLED macro.

zou3519 committed:
Update on "More named inference rules for pointwise unary ops"

More named inference rules for pointwise unary ops

Featuring:
- digamma, erfinv, lgamma, polygamma, sigmoid, tanh, zero_

Test Plan:
- `python test/test_namedtensor.py -v` [namedtensor ci]

gh-metadata: pytorch pytorch 22308 gh/zou3519/65/head
